ghostTemplate.yaml

```diff
@@ -0,0 +1,381 @@
+          Action: sts:AssumeRole
+  ghostAPITaskRoleAurora:
+    Type: AWS::IAM::Role
+    Condition: GhostAuroraDB
+    Properties:
+      RoleName: !Sub ${AWS::StackName}-ghostAPI-task
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          Effect: Allow
+          Principal:
+            Service: ecs-tasks.amazonaws.com
+          Action: sts:AssumeRole
+      Policies:
+        - PolicyName: SecretPolicy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - secretsmanager:GetSecretValue
+                Resource:
+                  - !Ref ghostDBRootUserSecret
+        - PolicyName: StackeryPolicy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - rds-data:BatchExecuteStatement
+                  - rds-data:BeginTransaction
+                  - rds-data:CommitTransaction
+                  - rds-data:ExecuteStatement
+                  - rds-data:RollbackTransaction
+                Resource:
+                  - !Sub "arn:aws:rds:${AWS::Region}:${AWS::AccountId}:cluster:${ghostDB}"
+        - PolicyName: AccessSecrets
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - secretsManager:GetSecretValue
+                Resource: !Sub "arn:${AWS::Partition}:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:/${EnvironmentTagName}/*"
+  ghostAPILogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /stackery/task/${AWS::StackName}-ghostAPI/logs
+  ghostDB:
+    Type: AWS::RDS::DBCluster
+    Condition: GhostAuroraDB
+    Properties:
+      BackupRetentionPeriod: 1
+      DBSubnetGroupName: !Ref ghostDBSubnetGroup
+      Engine: aurora
+      EngineMode: serverless
+      MasterUsername: !Sub '{{resolve:secretsmanager:${ghostDBRootUserSecret}:SecretString:username}}'
+      MasterUserPassword: !Sub '{{resolve:secretsmanager:${ghostDBRootUserSecret}:SecretString:password}}'
+      EnableHttpEndpoint: true
+      ScalingConfiguration:
+        AutoPause: true
+        MinCapacity: 1
+        MaxCapacity: 2
+      StorageEncrypted: true
+      VpcSecurityGroupIds:
+        - !Ref ghostDBSecurityGroup
+  ghostDBRootUserSecretAttachment:
+    Type: AWS::SecretsManager::SecretTargetAttachment
+    Condition: GhostAuroraDB
+    Properties:
+      SecretId: !Ref ghostDBRootUserSecret
+      TargetId: !Ref ghostDB
+      TargetType: AWS::RDS::DBCluster
+  ghostDBSubnetGroup:
+    Type: AWS::RDS::DBSubnetGroup
+    Condition: GhostAuroraDB
+    Properties:
+      DBSubnetGroupDescription: !Sub
+        - Stack ${StackTagName} Environment ${EnvironmentTagName} Database ${ResourceName} VPC Subnets
+        - ResourceName: ghostDB
+      SubnetIds: !Ref DefaultVPCSubnets
```

Comment:
Like with the load balancer targets, I think it's fine for this to have all the VPC subnets, but worth considering if it should be scoped to just the first two. I can't remember how aurora cluster DBs are set up.

Actually, IIRC you may be dropping this entirely in favor of ECS. If so, disregard :).

Reply:
Yeah, we should have a look at that too. I'd like your advice on how to properly scope the security-groups for private VPC-links. One's required, but I'm a little confused on its utility.